### ISSUE TITLE
(Feature request) Allow an array of animations

### ISSUE THREAD

**Issue by JoseExposito** [pixijs/pixi-tilemap#85]
Hi!

First at all, fantastic project. Thanks a lot for your work.

I'm building a Tiled (TMX) based small game and I'm using pixi-tilemap as renderer. I hope I'll be able to release the Tiled wrapper as a package soon.

I'm implementing animations and I'm facing an issue with the way Tiled allows to create them.
Long story short, Tiled gives you the flexibility to attach to any tile an animation that could start and end in any tile.

However, with pixi-tilemap I can only use `tileAnimX` and `tileAnimY`. This is OK if the animation starts in the attached tile and only has 2 frames, but, if the animation has 3 frames it will fail.
Basically this forces animation tiles to be in the same row/column.

It will be great if an API similar to this one was available:

```
tileAnim(steps: { offset: number, count: number }[]) // even time: number could be fantastic!
```

This will allow us to select any tile in the tileset.

I tried to implement it my self, and it is relatively easy to add this information to `RectTileLayer` but I have no idea about how to pass this information to the `RectTileShader` (I have never look into a shader before :S)

Thank you very much in advance,
Jose

**Comment by ivanpopelyshev** [comment]
the discussion was here: https://github.com/pixijs/pixi-tilemap/issues/76
you can have as many frames as you want, but they have to be consequtive X or Y coords, cant just take random frames
we need special shader that takes extra info from uniforms, its not that easy - i just dont have time to make this experiment  because there are 10 other pixi plugins im working on

**Comment by JoseExposito** [comment]
Wow, that's a lot of code to maintain! Actually it is not that big of a limitation, if that much work is involved it doesn't worth it.

Thanks again!
